rpcs/ledgerService.go

```diff
@@ -229,10 +230,37 @@ func (ls *LedgerService) ListenForCatchupReq(reqs <-chan network.IncomingMessage
 // a blocking function for handling a catchup request
 func (ls *LedgerService) handleCatchupReq(ctx context.Context, reqMsg network.IncomingMessage) {
 	var res WsGetBlockOut
+	target := reqMsg.Sender.(network.UnicastPeer)
+	var respTopics network.Topics
+
+	if target.Version() == "2.1" {
+		defer func() {
+			target.Respond(ctx, reqMsg, respTopics)
+		}()
+
+		topics, err := network.UnmarshallTopics(reqMsg.Data)
+		if err != nil {
+			topics = network.Topics{
+				network.MakeTopic(network.ErrorKey, []byte(err.Error()))}
+			return
+		}
+		roundBytes, found := topics.GetValue(network.RoundKey)
+		if !found {
+			topics = network.Topics{
+				network.MakeTopic(network.ErrorKey,
+					[]byte("LedgerService handleCatchupReq: round number topic is missing"))}
+			return
+		}
+		requestType, _ := topics.GetValue(network.RequestDataTypeKey)
+		round, _ := binary.Uvarint(roundBytes)
```

Comment:
You need to have error checking here to avoid potential attack vector.
( both requestType and round )